docs/protocol.md

```diff
@@ -662,7 +673,233 @@ Response (in case of success)
+                    "txbTmVMwCDkZdoaAiEYfAKozVizZzkeMkeOlzq5qMlg="
+                ]
+            }
+        ]
+    }
+}
+
+```
+
+The heartbeat can then trigger the internal logic of a server: when it receives messages ids that it has not already delivered it will execute a getMessagesById RPC. However, that is just a consequence of that specific condition, normally the heartbeat is just a way to communicate to others that the sender is alive, thus it doesn't expect any answer back.
+
+<details>
+<summary>
+💡 See the full specification
+</summary>
+  
+```json5
+// ../protocol/query/method/heartbeat.json
+
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/dedis/popstellar/master/protocol/query/method/heartbeat.json",
+    "description": "Heartbeat message",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "method": {
+            "description": "[String] operation to be performed by the query",
+            "const": "heartbeat"
+        },
+        "params": {
+            "description": "[Array] of objects containing each a channel and the list of ids from that channel",
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "maps": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": {
+                            "channel": {
+                                "description": "[String] name of the channel",
+                                "type": "string",
+                                "pattern": "^/root(/[^/]+)*$"
+                            },
+                            "ids": {
+                                "description": "[Array] of message_ids corresponding to that channel",
+                                "type": "array",
+                                "items": {
+                                    "type": "string"
+                                }
+                            }
+                        },
+                        "required": ["channel", "ids"]
+                    }
+                }
+            },
+            "required": ["maps"]
+        },
+        "jsonrpc": {
+            "$comment": "Defined by the parent, but needed here for the validation"
+        },
+
+        "id": {
+            "type": "integer"
+        }
+    },
+    "required": ["method", "params", "id", "jsonrpc"]
+}
+
+```
+</details>
+
+## Retrieving messages from server using ids 
+
+🧭 **RPC Message** > **Query** > **getMessagesById**
   
+This call is generally triggered upon the reception of a hearbeat message, in case the server hasn't already delivered some of the messages contained in such heartbeat but it could also used in other contexts. The purpose is to request missed messages so far.
```

Comment:
```suggestion
This call is generally triggered upon the reception of a hearbeat message, in case the server hasn't already received some of the messages contained in such heartbeat but it could also used in other contexts. The purpose is to request missed messages so far.
```
?

I would swap the two sentences, i.e. first describe what it does and then why/when it is used.

Also what message is used to send the answer to a `getMessagesById` query?

Reply:
The answer for that query is in getMessagesById_ans.json, it's basically the same as the heartbeat but with messages instead of ids